### PR TITLE
Issue 134 removing the no longer needed patch + adding integration tests

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Looks like the patch is no longer needed. Tested with maven 3.2.5 as well as 3.6.3. Added integration tests which test against the case described in the issue.

Some explanation.

The removed method extracted the imported dependencies from the raw model, which were then used *in addition to* dependencies from the imported boms, which were added by Maven.

But, because the dependencies added by Maven are not present in the POM to be updated, processing them is simply wasted resources. Instead, it's much better to just process the raw model.

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
Waiting for #476

**Comment by ajarmoniuk** [comment]
Looks like infrastructure problems again.

Could you please relaunch the failed jobs?

**Comment by ajarmoniuk** [comment]
@slawekjaranowski please review

**Comment by ajarmoniuk** [comment]
Also resolves #743

**Comment by ajarmoniuk** [comment]
Resolves: #743, #308, #309, #346, #395

File: versions-maven-plugin/src/it/it-unlock-snapshots-issue-134/pom.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  <version>1.0</version>
+  <packaging>pom</packaging>
+  <name>Check if boms are upated</name>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.codehaus.plexus</groupId>
+        <artifactId>plexus-utils</artifactId>
+        <version>1.5.1-20090311.051742-1</version>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-unlock-snapshots-issue-134/pom.xml:16]: Even if today such version of snapshot exists - can be removed, why not use some of dumy dependencies?

File: versions-maven-plugin/src/it/it-use-dep-version-issue-134/pom.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  <artifactId>it-use-releases-issue-134</artifactId>
+  <version>1.0</version>
+  <packaging>pom</packaging>
+  <name>Check if boms are upated</name>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.apache.maven</groupId>
+        <artifactId>maven-parent</artifactId>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-use-dep-version-issue-134/pom.xml:15]: why not use some of dumy dependencies?

File: versions-maven-plugin/src/it/it-unlock-snapshots-issue-134/pom.xml (hunk 1 above)
Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-unlock-snapshots-issue-134/pom.xml:16]: I coundt reproduce it with dummy dependencies. I did use dummy dependencies wherever I could.

Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-unlock-snapshots-issue-134/pom.xml:16]: As an alternative or a temporary workaround, I can remove these tree integration tests using non-dummy artifacts. Should I?

File: versions-maven-plugin/src/it/it-use-dep-version-issue-134/pom.xml (hunk 2 above)
Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-use-dep-version-issue-134/pom.xml]: Fixed

File: versions-maven-plugin/src/it/it-unlock-snapshots-issue-134/pom.xml (hunk 1 above)
Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-unlock-snapshots-issue-134/pom.xml]: Fixed

**Comment by ajarmoniuk** [comment]
@slawekjaranowski please review

**Comment by slawekjaranowski** [comment]
> Waiting for #476

rather replace - not waiting - I'm right?

**Comment by ajarmoniuk** [comment]
> > Waiting for #476
> 
> rather replace - not waiting - I'm right?

Not anymore. I meant to wait for the outcome of the discussion there.

**Pull request merged**